module/VuFind/src/VuFind/Controller/BrowseController.php

```diff
@@ -340,7 +340,9 @@ public function tagAction()
                     // Note -- this does not need to be escaped because
                     // $params['query'] has already been validated against
                     // the getAlphabetList() method below!
-                    $tags = $tagTable->matchText($params['query']);
+                    $where = [' LOWER(t.tag) LIKE :text AND rt.resource is NOT NULL '];
+                    $parameters['text'] = $params['query'] . '%';
+                    $tags = $tagService->getTagList(where: $where, parameters: $parameters);
```

Comment:
@skellamp, I think it might be worthwhile to move this logic into a `matchText` method in the tag service, because I notice that there's currently a second call to matchText that hasn't been ported over yet -- please take a look at the \VuFind\Autocomplete\Tags class. I don't think this should be too difficult to convert.

To test the tag autocomplete class, just put some tags on a record, and then next to the main search box, change the type drop-down to "Tag" -- now, when you type at least three letters, the autocomplete should suggest existing tags that start with those letters.

Reply:
Also note that the tag autocomplete class does have a corresponding unit test, so we'll need to adjust that to match. Right now, the class is using the default factory and relying on the DbTableAwareTrait to get its dependencies automatically injected. If it would make things easier, please feel free to create a TagFactory that explicitly injects the TagService -- you don't have to do things this way if you don't want to, but I think it will make the testing slightly less complicated, and the code's dependency injection more explicit.